### PR TITLE
Settings with comments

### PR DESCRIPTION
**Pull request by afshin**
- [x] Allows arbitrary serializable data being stored in the state database.
- [x] Adds support for JSON with comments raw settings being loaded from the back-end in the setting registry. (Depends on https://github.com/jupyterlab/jupyterlab_launcher/pull/25)
- [x] Individual settings can still be updated/saved without compromising the JSON comments.
- [x] Inspection handlers are now instantiated with an `IDataConnector`, which means that arbitrary sources can populate an inspector, whether it is the global inspector or a local instance (as in the case of the raw settings editor).
- [x] Completion handlers are now instantiated with an `IDataConnector`, which means that arbitrary sources can populate text completion. 

settings-with-comments

### CODE REVIEW

File: packages/settingeditor/src/plugineditor.ts [hunk 1]
```diff
@@ -205,10 +231,45 @@ namespace PluginEditor {
    */
   export
   interface IOptions {
+    /**
+     * The toolbar commands and registry for the setting editor toolbar.
+     */
+    commands: {
```
Comment (blink1073) [inline packages/settingeditor/src/plugineditor.ts]: misaligned and missing new line

File: packages/settingeditor/src/settingeditor.ts [hunk 2]
```diff
@@ -335,6 +412,31 @@ namespace SettingEditor {
    */
   export
   interface IOptions {
+    /**
+     * The toolbar commands and registry for the setting editor toolbar.
+     */
+    commands: {
```
Comment (blink1073) [inline packages/settingeditor/src/settingeditor.ts]: indent and new line

File: packages/settingeditor/src/plugineditor.ts (hunk 1 above)
Comment (afshin) [reply packages/settingeditor/src/plugineditor.ts]: `commands` is an object that is a child attribute of the `IOptions`

File: packages/settingeditor/src/settingeditor.ts (hunk 2 above)
Comment (afshin) [reply packages/settingeditor/src/settingeditor.ts]: Ditto.

**Comment by blink1073** [comment]
Very nice, thanks!

**Pull request merged**